packages/components/src/components/domainproperties/samples/SampleTypePropertiesPanel.tsx

```diff
@@ -107,34 +135,68 @@ class SampleTypePropertiesPanelImpl extends React.PureComponent<Props & Injected
         onAddParentAlias(newId, newParentAlias);
     };
 
-    renderAddEntityHelper = ():any => {
+    renderAddEntityHelper = (parentageLabel?: string):any => {
+        const msg = parentageLabel ? PARENT_ALIAS_HELPER_TEXT.replace("parentage", parentageLabel) : PARENT_ALIAS_HELPER_TEXT;
         return (
             <>
                 <span>
-                    <p>{PARENT_ALIAS_HELPER_TEXT}</p>
+                    <p>{msg}</p>
                     <p>{helpLinkNode(DERIVE_SAMPLES_ALIAS_TOPIC, "More info")}</p>
                 </span>
             </>
         );
     };
 
-    renderParentAliases = () => {
-        const {model, parentOptions, updateDupeParentAliases} = this.props;
+    renderParentAliases = (includeSampleSet: boolean, includeDataClass: boolean) => {
+        const {model, parentOptions, updateDupeParentAliases, sampleAliasCaption, sampleTypeCaption, dataClassAliasCaption, dataClassTypeCaption, dataClassParentageLabel } = this.props;
         const {parentAliases} = model;
 
         if (!parentAliases || !parentOptions)
             return [];
 
-        return parentAliases.valueSeq().map((alias:IParentAlias) => {
+        let filteredParentAliases = OrderedMap<string, IParentAlias>();
+        let filteredParentOptions = Array<IParentOption>();
+        let aliasCaption;
+        let parentTypeCaption;
+
+        let helpMsg = undefined;
+        if (includeSampleSet && includeDataClass) {
+            filteredParentAliases = parentAliases;
+            filteredParentOptions = parentOptions;
+        }
+        else if (includeSampleSet) {
+            filteredParentAliases = parentAliases.filter(sampleSeAliasFilterFn) as OrderedMap<string, IParentAlias>;
+            filteredParentOptions = parentOptions.filter(sampleSetOptionFilterFn);
+            if (sampleAliasCaption)
+                aliasCaption = sampleAliasCaption;
+            if (sampleTypeCaption)
+                parentTypeCaption = sampleTypeCaption;
+        }
+        else if (includeDataClass) {
+            filteredParentAliases = parentAliases.filter(dataClassAliasFilterFn) as OrderedMap<string, IParentAlias>;
+            filteredParentOptions = parentOptions.filter(dataClassOptionFilterFn);
+            if (dataClassAliasCaption)
+                aliasCaption = dataClassAliasCaption;
+            if (dataClassTypeCaption)
+                parentTypeCaption = dataClassTypeCaption;
+
+            if (dataClassParentageLabel)
+                helpMsg = PARENT_ALIAS_HELPER_TEXT.replace("parentage", dataClassParentageLabel)
```

Comment:
This bit of substitution is specialized enough (and used in two places) that I might separate it into a method and maybe pass `dataClassParentageLabel` into `SampleSetParentAliasRow` and let it call the function.  Neither way is particularly elegant, though.

Reply:
Yeah neither way is very elegant... SampleSetParentAliasRow doesn't currently know anything about sample or dataclass, so I went with have it be the consumer of the already processed msg, to avoid adding complexity to it.